meerk40t/gui/materialmanager.py

```diff
@@ -0,0 +1,1216 @@
+        for i, key in enumerate(dictionary.keys()):
+            if i == n:
+                return key
+        return None
+
+    def get_nth_material(self, n=0):
+        return self.get_nth_dict_entry(self.material_list, n)
+
+    def get_nth_operation(self, n=0):
+        return self.get_nth_dict_entry(self.operation_list, n)
+
+    def on_share(self, event):
+        if self.active_material is None:
+            return
+
+        self.context.setting(str, "author", "")
+        last_author = self.context.author
+        if last_author is None:
+            last_author = ""
+        dlg = wx.TextEntryDialog(
+            self,
+            _(
+                "Thank you for your willingness to share your material setting with the MeerK40t community.\n" +
+                "Please provide a name to honor your authorship."
+            ), caption=_("Share material setting"),
+            value=last_author,
+        )
+        dlg.SetValue(last_author)
+        res = dlg.ShowModal()
+        last_author = dlg.GetValue()
+        dlg.Destroy()
+        if res == wx.ID_CANCEL:
+            return
+        self.context.author = last_author
+        # We will store the relevant section in a separate file
+        oplist, opinfo = self.context.elements.load_persistent_op_list(self.active_material)
+        if len(oplist) == 0:
+            return
+        opinfo["author"] = last_author
+        directory = get_safe_path(self.context.kernel.name)
+        local_file = os.path.join(directory, "op_export.cfg")
+        if os.path.exists(local_file):
+            try:
+                os.remove(local_file)
+            except (OSError, PermissionError):
+                return
+        settings = Settings(
+            self.context.kernel.name,
+            "op_export.cfg",
+            ignore_settings=False
+        )
+        opsection = f"{self.active_material} info"
+        for key, value in opinfo.items():
+            settings.write_persistent(opsection, key, value)
+        def _save_tree(name, op_list):
+            for i, op in enumerate(op_list):
+                if hasattr(op, "allow_save"):
+                    if not op.allow_save():
+                        continue
+                if op.type == "reference":
+                    # We do not save references.
+                    continue
+                section = f"{name} {i:06d}"
+                settings.write_persistent(section, "type", op.type)
+                op.save(settings, section)
+                try:
+                    _save_tree(section, op.children)
+                except AttributeError:
+                    pass
+        _save_tree(opsection, oplist)
+        settings.write_configuration()
+        # print ("Sharing")
+        try:
+            with open(local_file, "r") as f:
+                data = f.read()
+        except:
+            return
+        self.send_data_to_developers(local_file, data)
+
+    def send_data_to_developers(self, filename, data):
```

Comment:
This is weird to have here.

Reply:
The idea was to have an opportunity to share your settings with the mk community. It's a stub for the time being as the backend counterpart needs to be in place too.